src/config/config.cc

```diff
@@ -829,8 +829,8 @@ Status Config::Rewrite() {
   std::ifstream file(path_);
   if (file.is_open()) {
     std::string raw_line;
-    while (!file.eof()) {
-      std::getline(file, raw_line);
+    while (std::getline(file, raw_line)) {
+      if (file.eof()) break;
```

Comment:
If the text file has no newline after the last line, this line will be ignored.

e.g.
```
file.write("a\nb\nc"); # the `c` line will be ignored
```

Hence I think it is not a good change.